lib/libxdp/xsk.c

```diff
@@ -162,25 +162,19 @@ static void xsk_set_umem_config(struct xsk_umem_config *cfg,
 }
 
 static int xsk_set_xdp_socket_config(struct xsk_socket_config *cfg,
-				     const struct xsk_socket_config *usr_cfg)
+				     const struct xsk_socket_opts *opts)
 {
-	if (!usr_cfg) {
-		cfg->rx_size = XSK_RING_CONS__DEFAULT_NUM_DESCS;
-		cfg->tx_size = XSK_RING_PROD__DEFAULT_NUM_DESCS;
-		cfg->libbpf_flags = 0;
-		cfg->xdp_flags = 0;
-		cfg->bind_flags = 0;
-		return 0;
-	}
+	__u32 libbpf_flags;
 
-	if (usr_cfg->libbpf_flags & ~XSK_LIBBPF_FLAGS__INHIBIT_PROG_LOAD)
+	libbpf_flags = OPTS_GET(opts, libbpf_flags, 0);
+	if (libbpf_flags & ~XSK_LIBBPF_FLAGS__INHIBIT_PROG_LOAD)
```

Comment:
Please use LIBXDP instead of LIBBPF in this whole function. Again, the LIBBPF stuff in the interface is only there for some old libbpf compatibility. The local variable can be libxdp_flags too since we are in libxdp :-).